totalRP3/Modules/Register/Characters/RegisterRelations.lua

```diff
@@ -0,0 +1,348 @@
+end
+TRP3_API.register.relation.getRelationText = getRelationText;
+
+
+local function getRelationTooltipText(profileID, profile)
+	local description = getRelation(profileID).description or loc:GetText("REG_RELATION_" .. getRelation(profileID).id .. "_TT");
+	local player = getPlayerCompleteName(true);
+	local target = getCompleteName(profile.characteristics or EMPTY, UNKNOWN, true);
+	return description:format(player, target)
+end
+TRP3_API.register.relation.getRelationTooltipText = getRelationTooltipText;
+
+local function getRelationTexture(profileID)
+	return getRelation(profileID).texture;
+end
+TRP3_API.register.relation.getRelationTexture = getRelationTexture;
+
+local function getRelationColors(profileID)
+	local relation = getRelation(profileID);
+	return TRP3_API.CreateColorFromHexString(relation.color);
+end
+TRP3_API.register.relation.getRelationColors = getRelationColors;
+
+local function getColor(relation)
+	return TRP3_API.CreateColorFromHexString('#'..getRelationInfo(relation).color);
+end
+TRP3_API.register.relation.getColor = getColor;
+
+--*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+-- INIT
+--*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+
+local function checkRelationUse()
+	local relationList = getRelationList()
+	for _, relation in pairs(relationList) do
+		relation.inUse = false
+	end
+	local profiles = getProfiles()
+	for _, profile in pairs(profiles) do
+		local relations = get("relation", profile)
+		if not relations then
+			relations = {}
+		end
+		for _, relation in pairs(relations) do
+			getRelationInfo(relation).inUse = true
+		end
+	end
+end
+
+local draftRelationTexture;
+
+-- Init the relation popup
+local function initRelationEditor(relationID)
+	TRP3_RelationsList.Editor.Content.Title:SetText(loc.CO_RELATIONS_MENU_EDIT);
+	TRP3_RelationsList.Editor.Content.ID = relationID;
+
+	local relation = getRelationInfo(relationID)
+	draftRelationTexture = relation.texture;
+	-- set icon, name, description, color
+	TRP3_API.ui.frame.setupIconButton(TRP3_RelationsList.Editor.Content.Icon, relation.texture or TRP3_InterfaceIcons.ProfileDefault);
+	TRP3_RelationsList.Editor.Content.Icon:SetScript("OnClick", function()
+		TRP3_API.popup.showPopup(TRP3_API.popup.ICONS, nil, {function(icon)
+			draftRelationTexture = icon;
+			TRP3_API.ui.frame.setupIconButton(TRP3_RelationsList.Editor.Content.Icon, icon or TRP3_InterfaceIcons.Default);
+		end, nil, nil, draftRelationTexture});
+	end);
+
+	local nameText = relation.name;
+	if not nameText then
+		nameText = loc:GetText("REG_RELATION_"..relation.id);
+	end
+	TRP3_RelationsList.Editor.Content.Name:SetText(nameText);
+
+	local descriptionText = relation.description;
+	if not descriptionText then
+		descriptionText = loc:GetText("REG_RELATION_" .. relation.id .. "_TT");
+	end
+	local playerIdentifier = "%%p";
+	local targetIdentifier = "%%t";
+	TRP3_RelationsList.Editor.Content.Description:SetText(descriptionText:gsub('%%1$s', playerIdentifier):gsub('%%2$s', targetIdentifier));
```

Comment:
The two gsubs can be merged into one using table-lookup replacements. Anything that matches the pattern but has no corresponding key in the table will be left unaltered.

```lua
descriptionText:gsub("%%.$s", { ["%1$s"] = "%p", ["%2$s"] = "%t" })
```